spec/javascripts/services/shared_service_spec.coffee

```diff
@@ -0,0 +1,34 @@
+# mock globals
+STATIC_ASSET_PATHS
+do ->
+  'use strict'
+  describe 'SharedService', ->
+    SharedService = undefined
+    httpBackend = undefined
+    $state = undefined
+    $window = undefined
+    $document = undefined
+    languageMap: {es: 'Spanish'}
+
+    beforeEach module('ui.router')
+    beforeEach module('dahlia.services', ($provide) ->
+    )
+
+    beforeEach inject((_SharedService_, _$httpBackend_, _$state_, _$window_, _$document_) ->
+      httpBackend = _$httpBackend_
+      $state = _$state_
+      $window = _$window_
+      $document = _$document_
+      SharedService = _SharedService_
+      return
+    )
+
+    describe 'getLanguageCode', ->
+      it 'returns the 2-letter code for the given language name', ->
+        code = SharedService.getLanguageCode('Spanish')
+        expect(code).toEqual 'es'
```

Comment:
@callaghanc Yay, `SharedService` tests! What about testing all languages by adding the rest to `languageMap` and then
```coffee
Object.keys(languageMap).forEach key ->
  language = languageMap[key]
  languageCode = SharedService.getLanguageCode(language)
  expect(languageCode).toEqual key
```

Reply:
sure, added testing of all languages to the language name and code function tests